llama-index-core/llama_index/core/workflow/service.py

```diff
@@ -0,0 +1,29 @@
+from typing import Dict, TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from .workflow import Workflow
+
+
+class ServiceNotFoundError(Exception):
+    """An error raised when the service manager couldn't find a certain service name."""
+
+
+class ServiceManager:
```

Comment:
We already discussed this in the PR, to recap: in core this is not expected to do much more than a dict, but most likely going to be needed in llama-agents